common/ScaLBL.cpp

```diff
@@ -2544,6 +2544,8 @@ void ScaLBL_Communicator::SendD3Q7AA(double *Aq, int Component) {
     //...Packing for x face(2,8,10,12,14)................................
     ScaLBL_D3Q19_Pack(2, dvcSendList_x, 0, sendCount_x, sendbuf_x,
                       &Aq[Component * 7 * N], N);
+
+    ScaLBL_DeviceBarrier();
     req1[0] =
```

Comment:
You should be able to
1) make  all the calls to `ScaLBL_D3Q19_Pack` in a row
2) make a single call to `ScaLBL_DeviceBarrier()`
3) all of the calls to `MPI_COMM_SCALBL.Isend` in a row
This way there is only one synchronization point, which should be faster.